client/dashboard/app/queries/site-backup-download.ts

```diff
@@ -0,0 +1,51 @@
+import { mutationOptions, queryOptions } from '@tanstack/react-query';
+import {
+	fetchSiteBackupDownloadProgress,
+	initiateSiteBackupDownload,
+	type DownloadProgress,
+	type DownloadConfig,
+} from '../../data/site-backup-download';
+import { queryClient } from '../query-client';
+
+/**
+ * Fetch the download progress for a site backup.
+ * @param siteId - The ID of the site to fetch download progress for.
+ * @param downloadId - The ID of the download to fetch progress for.
+ * @returns A promise that resolves to the download progress.
+ */
+export const siteBackupDownloadProgressQuery = ( siteId: number, downloadId: number ) =>
+	queryOptions( {
+		queryKey: [ 'site', siteId, 'backup', 'download', downloadId, 'progress' ],
+		queryFn: () => fetchSiteBackupDownloadProgress( siteId, downloadId ),
+		refetchInterval: ( query: { state: { data?: DownloadProgress } } ) => {
```

Comment:
Let's move this to the component; this is a business logic that's specific to that screen.